run.c

```diff
@@ -543,7 +547,11 @@ int main(int argc, char *argv[]) {
     int next;        // will store the next token in the sequence
     int token = 1;   // init with token 1 (=BOS), as done in Llama-2 sentencepiece tokenizer
     int pos = 0;     // position in the sequence
+    int bufferflush = 1; // buffer flush after token counter 
+    char outbuff[4096]; // used for output buffering              
```

Comment:
This should be static, of size `BUFSIZ`. Static array will be initialized with zeros automatically, not that it's actually needed (i.e. you can remove the memeset on the next line).

>A common error is setting the buffer of stdin or stdout to an array whose lifetime ends before the program terminates:

https://en.cppreference.com/w/c/io/setvbuf

```suggestion
    static char outbuf[BUFSIZ]; // used for output buffering              
```

Reply:
BUFSIZ differs from OS to OS & Distros and standard libs used. 

I built with two tool chains and I have binaries that report differing BUFSIZ:
BUFSIZ of 4096 with cosmo libc &
BUFSIZ of 8192, my OS default

Some OSs have only 1024 or even 512 BUFSIZ.

Reply:
>The default buffer size BUFSIZ is expected to be the most efficient buffer size for file I/O on the implementation,

https://en.cppreference.com/w/c/io/setvbuf

Reply:
Here we do not use the buffer for high performance IO, rather as a fast enough place to dump tokens instead of writing to screen. We wait till the buffer is filled up with desired number of tokens and then we print and flush. We try to do it as simple as possible without writing tons of code. This speeds up stuff on low end systems, less time wasted writing to screen.

Tokens can have varying length. The buffer must be larger than the total length of the tokens to be held. There were cases where output was corrupted due to lengthy and repetitive responses and a smaller buffer of 2048. A large enough size that I found works well is 4096.

Reply:
>The buffer must be larger than the total length of the tokens to be held. 

4096 does not guarantee it, not that I think this guarantee is necessary.

>There were cases where output was corrupted due to lengthy and repetitive responses and a smaller buffer of 2048. 

This must be a bug elsewhere. How? Like you've said, actual default buffer size can be smaller than that.